### DISCUSSION TITLE
How to change property type for migration, properly?

### DISCUSSION THREAD

**Discussion by NetOpWibby** [geldata/gel#4899] [Q&A] [answered]
I changed a boolean property to an `int16`. Here's what `edgedb migration edit --instance myInstance --database myDatabase` returns:

```
CREATE MIGRATION m1e5taupee7r2aoduprzzlhptczmr45xzqpgy3ewgjyomcgw6xlfkq
    ONTO m147brslrjen4ia2x3zvwtvkivz2hm57wgdm3gm6a4p3th6z4kb5na
{
  ALTER TYPE default::Registrar {
      ALTER PROPERTY enrolled {
          SET default := 0;
      };
  };
};
```

When attempting to update a document, I get `Error: Unknown numerical type: std::bool!`. Not sure what I'm doing wrong, docs are lacking information on migrating types. When searching Neeva I saw this issue: https://github.com/edgedb/edgedb/issues/3953.

**Comment by raddevon** [comment]
I tried this migration, and was able to pull it off successfully. Here's what I did.

I started with this schema:
```
module default {
  type Registrar {
    property enrolled -> bool;
  }
}
```

I inserted a few objects with different values for `enrolled`. Then, I updated the schema to this:
```
module default {
  type Registrar {
    property enrolled -> int64;
  }
}
```

I ran this:
```bash
edgedb migration create && edgedb migrate
```

When it prompted me to cast the values in the revised property, I gave it this expression:
```
0 if .enrolled = false else 1
```

I ran a query to confirm all `false` values had been changed to `0` and `true` to `1`.

After this, I was able to update objects. I ran this query to test:
```
update Registrar filter .enrolled = 1 set {enrolled := 2};
```

and verified all of my `1`s were now `2`s.

**Reply by NetOpWibby** [reply]
I'm still getting `Unknown numerical type: std::bool!`

**Reply by raddevon** [reply]
You did all the steps described above, but you get that error? At what point do you get the error?

**Comment by NetOpWibby** [comment, marked as answer]
In Discord, @raddevon helped me realize I needed to re-run the query builder.

```sh
deno run -A --unstable https://deno.land/x/edgedb@v1.0.1/generate.ts edgeql-js --instance myInstance --database myDatabase
```
